File: minCosts.py
# https://leetcode.com/problems/minimum-cost-to-reach-every-position/

class Solution(object):
	def minCosts(self, cost):
		m = float('inf')
		res = []
		for c in cost:
			m = min(m, c)
			res.append(m)
		return res

File: minCosts.java
// https://leetcode.com/problems/minimum-cost-to-reach-every-position/

class Solution {
	public int[] minCosts(int[] cost) {
		int m = Integer.MAX_VALUE;
		int[] res = new int[cost.length];
		for (int i = 0; i < cost.length; i++) {
			m = Math.min(m, cost[i]);
			res[i] = m;
		}
		return res;
	}
}